### DISCUSSION TITLE
Webview/Mini-browser Endpoints

### DISCUSSION THREAD

**Discussion by mrora** [eclipse-theia/theia#10699] [General] [answered]
i have a question about webviews/mini-browsers. quite a bit of work has gone into isolating the iframes for these widgets from the rest of theia. when the `WebviewExternalEndpoint` or `MiniBrowserEndpoint` is overridden, a security warning is shown that explains that this change could be lead to a security vulnerability. i want to better understand the implications of this.

my understanding is that if those endpoint patterns are changed such that the origin of the webview/mini-browser iframe is the same as the containing frame's origin, then `localStorage`, `cookies`, etc. from the containing frame would be accessible to the webview/mini-browser (or vice-versa). is that right?

is the concern here that a specially crafted plugin/extension could take advantage of the missing cross-origin isolation? are there more concerns to be aware of for this scenario?

thanks!

**Comment by msujew** [comment]
cc @paul-marechal, since you implemented it in https://github.com/eclipse-theia/theia/pull/9563.

**Comment by paul-marechal** [comment, marked as answer]
> my understanding is that if those endpoint patterns are changed such that the origin of the webview/mini-browser iframe is the same as the containing frame's origin, then `localStorage`, `cookies`, etc. from the containing frame would be accessible to the webview/mini-browser (or vice-versa). is that right?

Correct. In addition to isolating the extension's browsing context from the main Theia application, it also isolate extensions from each other, avoiding potential corruption of browser storage by having everyone share the same context.

Cookies might also carry authentication information, so there might be setups that benefit from not sharing everything with third-party programs downloaded from public registries.

> are there more concerns to be aware of for this scenario?

There might be that I am unaware of. I see this isolation as a "better safe than sorry" strategy.

**Reply by mrora** [reply]
thanks for the info, @paul-marechal!
